solidity/contracts/token/bridge/EverclearTokenBridge.sol

```diff
@@ -0,0 +1,235 @@
+        address _mailbox,
+        IEverclearAdapter _everclearAdapter
+    ) HypERC20Collateral(erc20, _scale, _mailbox) {
+        everclearAdapter = _everclearAdapter;
+    }
+
+    /**
+     * @notice Initializes the contract with required parameters
+     * @dev Sets up the mailbox client and approves the Everclear adapter to spend tokens
+     * @param _hook The address of the post-dispatch hook contract
+     * @param _interchainSecurityModule The address of the interchain security module
+     * @param _owner The address that will own this contract
+     */
+    function initialize(
+        address _hook,
+        address _interchainSecurityModule,
+        address _owner
+    ) public override initializer {
+        _MailboxClient_initialize(_hook, _interchainSecurityModule, _owner);
+        wrappedToken.approve(address(everclearAdapter), type(uint256).max);
+    }
+
+    /**
+     * @notice Sets the fee parameters for Everclear bridge operations
+     * @dev Only callable by the contract owner
+     * @param _fee The fee amount to charge users for bridge operations
+     * @param _deadline The deadline timestamp for fee parameter validity
+     * @param _sig The signature for fee validation from Everclear
+     */
+    function setFeeParams(
+        uint256 _fee,
+        uint256 _deadline,
+        bytes calldata _sig
+    ) external onlyOwner {
+        feeParams = IEverclearAdapter.FeeParams({
+            fee: _fee,
+            deadline: _deadline,
+            sig: _sig
+        });
+        emit FeeParamsUpdated(_fee, _deadline);
+    }
+
+    /**
+     * @notice Sets the output asset address for a destination domain
+     * @dev Only callable by the contract owner
+     * @param _destination The destination domain ID
+     * @param _outputAsset The output asset address on the destination chain (as bytes32)
+     */
+    function setOutputAsset(
+        uint32 _destination,
+        bytes32 _outputAsset
+    ) external onlyOwner {
+        outputAssets[_destination] = _outputAsset;
+        emit OutputAssetSet(_destination, _outputAsset);
+    }
+
+    /**
+     * @notice Sets multiple output assets in a single transaction for gas efficiency
+     * @dev Only callable by the contract owner. Arrays must be the same length
+     * @param _destinations Array of destination domain IDs
+     * @param _outputAssets Array of output asset addresses on destination chains
+     */
+    function setOutputAssetsBatch(
+        uint32[] calldata _destinations,
+        bytes32[] calldata _outputAssets
+    ) external onlyOwner {
+        require(
+            _destinations.length == _outputAssets.length,
+            "ETB: Length mismatch"
+        );
+
+        uint256 len = _destinations.length;
+
+        for (uint256 i = 0; i < len; ) {
+            outputAssets[_destinations[i]] = _outputAssets[i];
+            emit OutputAssetSet(_destinations[i], _outputAssets[i]);
+            unchecked {
+                ++i;
+            }
```

Comment:
I think this type of gas optimization is not worth the readability hit
this function's gas cost is not incurred by users

Reply:
Done